src/HexagonOptimize.cpp

```diff
@@ -679,6 +681,27 @@ class OptimizePatterns : public IRMutator {
                 expr = mutate(op->a + neg_b);
                 return;
             } else {
+                // No widening subtracts for the expressions of the form:
```

Comment:
I think it would be better to implement this by adding signed widening subtract functions (similar to halide.hexagon.sub_vuh.vub.vub) with implementations in hvx_64.ll and hvx_128.ll (using vmpa), and then just use a regular pattern to match it.

BTW, I'm hoping to do major refactoring/simplification of the vrmpy and vtmpy implementation at some point.